riscv_analysis/src/parser/lexer.rs

```diff
@@ -128,52 +156,108 @@ impl Lexer {
         }
     }
 
+    /// Lex a unicode escape code.
+    ///
+    /// Returns None if the code doesn't define a valid unicode character.
```

Comment:
I would add a comment that we are lexing the ASCII representations into a single unicode character.